server/irmaserver/sessions.go

```diff
@@ -138,58 +154,188 @@ func (s *memorySessionStore) deleteExpired() {
 
 	// Using a write lock, delete the expired sessions
 	s.Lock()
-	for _, token := range expired {
-		session := s.requestor[token]
+	for _, Token := range expired {
+		session := s.requestor[Token]
 		if session.sse != nil {
-			session.sse.CloseChannel("session/" + session.token)
-			session.sse.CloseChannel("session/" + session.clientToken)
+			session.sse.CloseChannel("session/" + session.Token)
+			session.sse.CloseChannel("session/" + session.ClientToken)
 		}
-		delete(s.client, session.clientToken)
-		delete(s.requestor, token)
+		delete(s.client, session.ClientToken)
+		delete(s.requestor, Token)
 	}
 	s.Unlock()
 }
 
+// MarshalJSON marshals sessionData to be used in the Redis in-memory datastore.
+func (s *sessionData) MarshalJSON() ([]byte, error) {
+	return json.Marshal(*s)
+}
+
+// UnmarshalJSON unmarshals sessionData.
+func (s *sessionData) UnmarshalJSON(data []byte) error {
+	type rawSessionData sessionData
+
+	var temp struct {
+		Rrequest *json.RawMessage `json:",omitempty"`
```

Comment:
adjusted.

For consistency the other temp struct used for unmarshaling the LogEntry was also adjusted. @ivard Would you mind to have a special look at https://github.com/privacybydesign/irmago/pull/149/commits/330078e38a0664222da84bac77c0fad2df703a37  ?

Reply:
There are certain LogEntry's for irmaclient (commit 330078e) that have no `serverName` at all (I know, that should be fixed but that is a story on Gitlab). Are we sure that this parses when `serverName` is `nil`. Otherwise the app will start crashing.

Reply:
It should: the underlying type of `json.RawMessage` is just `[]byte`, which can be `nil` just like `*json.RawMessage` can. But we can and should test this to be sure.

Reply:
The docs of the `len(v Type)` function states:
```
//	Array: the number of elements in v.
//	Pointer to array: the number of elements in *v (even if v is nil).
```
`json.RawMessage` has an underlaying byte[] type. So no matter if the byte array is nil, empty or even if we'd use a pointer to the point array, we're good :)